src/rebar_packages.erl

```diff
@@ -254,6 +257,19 @@ update_package(Name, RepoConfig=#{name := Repo}, State) ->
             fail
     end.
 
+get_package_repo_config(RepoConfig=#{mirror_of := Repo}) ->
+    get_package_repo_config(maps:remove(mirror_of, RepoConfig#{name => Repo}));
+get_package_repo_config(RepoConfig=#{read_key := Key}) ->
+    get_package_repo_config(maps:remove(read_key, RepoConfig#{repo_key => Key}));
+get_package_repo_config(RepoConfig) ->
+    RepoConfig.
+
+unverified_repo_message() ->
+    "Fetched deprecatated registry record version from repo ~ts, for security " ++
+        "reasons this registry version is no longer supported. The repository " ++
+        "you are using should update to fix the security reason. Set " ++
+        "REBAR_NO_VERIFY_REPO_ORIGIN=1 to disable this check.".
```

Comment:
You can drop the `++` here since Erlang automatically appends literal string declarations at compile time.

I'd change the error to something like:

> The registry repository ~ts uses a record format that has been deprecated for security reasons. The repository should be updated in order to be safer. You can disable this check by setting REBAR_NO_VERIFY_REPO_ORIGIN=1"

Which I think might be clearer